### ISSUE TITLE
[Feature Request] Directory Listing Control and Initial Directory Configuration for S3 Buckets"

### ISSUE THREAD

**Issue by mauroalexandre** [mickael-kerjean/filestash#744]
Hello, congratulations on the fantastic project. 

I would like to ask a question. Unfortunately, S3 does not allow through IAM to prevent a user from seeing the list of directories in an S3 Bucket. Therefore, when the user logs in, they land on a directory listing they may not always have access to. I have two questions:

**1 -** Is it possible to address this situation through configurations?

**2 -** If not, may I implement and submit a Pull Request to enable configuring a 'start directory' so that the user is directed to a specific directory after login? Would this feature be adherent to the project?

Thank you.

**Comment by mickael-kerjean** [comment]
Yes, this is what the "path" field is about:

![screenshot-20240919-010904Z-all](https://github.com/user-attachments/assets/b52e57ef-4b47-40ca-90c8-1879c3e44bd8)

If you put: /bucket/ it will chroot the user within a bucket and won't be running any bucketlisting operation as it only happen at the root

**Comment by mauroalexandre** [comment]
@mickael-kerjean 

Cool! But would it be possible to configure this at the connection level? So that this parameterization is not user-dependent?

**Comment by mickael-kerjean** [comment]
yes, this is precisely what the auth middleware from the admin console is for. You can build those path dynamically based on whatever attribute is being passed through or even build rules around that as well. I've revamped the doc in there https://www.filestash.app/docs/install-and-upgrade/ 2 days ago to make those sort of things clearer but not sure if that's really good yet. Happy to get feedbacks as if it's not 100% clear, it means it got to be improved a lot more
